samples/collect_data/collect_data.cpp

```diff
@@ -0,0 +1,196 @@
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2021 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+// Application for testing non-CPU-intensive features of BOINC.
+// TODO: make it exit after a certain amount of time
+
+#include <sstream>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <chrono>
+#include <thread>
+#include <sockpp/socket.h>
+#include <sockpp/tcp_connector.h>
+#if !_WIN32
+    #include <sockpp/unix_connector.h>
+#endif
+
+void my_sleep(std::chrono::nanoseconds nanoseconds) {
+    std::this_thread::sleep_for(nanoseconds);
+}
+
+sockpp::connector* get_connected(sockpp::connector* array_connection[], int arrSize) {
+    if (NULL == array_connection || 0 == arrSize) {
+        return NULL;
+    }
+    for (int i = 0; i < arrSize; ++i) {
+        sockpp::connector* conn = array_connection[i];
+        if (conn != NULL && conn->is_connected()) {
+            return conn;
+        }
+    }
+    return NULL;
+}
+
+bool is_connected(sockpp::connector* array_connection[], int arrSize) {
+    if (NULL == array_connection || 0 == arrSize) {
+        return NULL;
+    }
+    for (int i = 0; i < arrSize; ++i) {
+        sockpp::connector* conn = array_connection[i];
+        if (conn != NULL && conn->is_connected()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+int main(int, char**) {
+    //boinc_init();
+
+    sockpp::socket_initializer sockInit;
+    sockpp::tcp_connector tcp_conn;
+    sockpp::connector* array_connection[] = { &tcp_conn, NULL };
+#if !_WIN32
+    sockpp::unix_connector unix_conn;
+    array_connection[1] = &unix_conn;
+#endif
+    sockpp::connector* conn = NULL;
+    const int arrSize = sizeof(array_connection) / sizeof(array_connection[0]);
+    const bool isDebug = false;
+    const int16_t port = 31416;
+    const int buffer_size = 4096;
+    const int wake_minutes = 30;
+    const int exec_hourly = 5;
+    int suffix_file_idx = 0;
+    const std::string android_address = "edu_berkeley_boinc_client_socket";
+    const std::string unix_address = "boinc_socket";
+    const std::string tcp_address = "localhost";
+    const std::string prefix_file_name = "boinc";
+    const std::chrono::minutes wake_every_minutes = std::chrono::minutes(wake_minutes);
```

Comment:
This is available starting from c++11 only
https://www.cplusplus.com/reference/chrono/